src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/impl/PaymentWorksVendorDaoJdbc.java

```diff
@@ -10,79 +13,132 @@
+            sqlFactory.appendSql(", vndr_hdr_gnrtd_id = ?", vendorHeaderGeneratedIdentifier);
         }
         if (ObjectUtils.isNotNull(vendorDetailAssignedIdentifier)) {
-            sql.append(", vndr_dtl_asnd_id = '" + vendorDetailAssignedIdentifier.intValue() + "'");
+            sqlFactory.appendSql(", vndr_dtl_asnd_id = ?", vendorDetailAssignedIdentifier);
         }
         if (ObjectUtils.isNotNull(kfsAchDocumentNumber)) {
-            sql.append(", paat_fdoc_nbr = '" + kfsAchDocumentNumber + "'");
+            sqlFactory.appendSql(", paat_fdoc_nbr = ?", kfsAchDocumentNumber);
         }
-        sql.append(" where id = '" + id + "'");
-        
-        return sql.toString();
+        sqlFactory.appendSql(" where id = ?", id);
+
+        return sqlFactory.getPreparedSqlStatement();
     }
 
+    private class ParameterizedSqlFactory {
+        private StringBuilder sqlStringBuilder;
+        private ArrayList parameters;
+
+        public ParameterizedSqlFactory(String initialSql) {
+            sqlStringBuilder = new StringBuilder(initialSql);
+            parameters = new ArrayList();
+        }
+
+        public void appendSql(String sql) {
+            sqlStringBuilder.append(sql);
+        }
+
+        public void appendSql(String sql, Object parameter) {
+            appendSql(sql);
+            parameters.add(parameter);
+        }
+
+        public PreparedStatement getPreparedSqlStatement() throws SQLException {
+            PreparedStatement preparedSqlStatement = getConnection().prepareStatement(sqlStringBuilder.toString());
+            for (int parameterIndex=1; parameterIndex<=parameters.size(); ++parameterIndex) {
+                addParameterToPreparedStatement(preparedSqlStatement, parameterIndex, parameters.get(parameterIndex - 1));
+            }
+            return preparedSqlStatement;
+        }
+
+        private void addParameterToPreparedStatement(PreparedStatement preparedSqlStatement, int parameterIndex, Object parameterValue) throws SQLException {
+            if (parameterValue instanceof Integer) {
+                preparedSqlStatement.setInt(parameterIndex, (Integer) parameterValue);
+            }
+            else {
+                preparedSqlStatement.setString(parameterIndex, parameterValue.toString());
+            }
+        }
+
+        public String toString() {
+            String effectiveSql = sqlStringBuilder.toString();
+            int parameterIndex = parameters.size();
+
+            while (parameterIndex > 0) {
+                int lastQIndex = effectiveSql.lastIndexOf("?");
+                Object parameter = parameters.get(parameterIndex - 1);
+                String parameterSql = parameter instanceof String ? "'" + parameter.toString() + "'" : parameter.toString();
+                effectiveSql = effectiveSql.substring(0, lastQIndex) + parameterSql + effectiveSql.substring(lastQIndex + 1);
```

Comment:
This loop is building a new full SQL string upon each iteration. I would recommend using some other alternative that can operate on the string more efficiently (like using a StringBuilder, or creating an intermediate string that replaces all the question marks with placeholders that can be processed by String.format() or a similar substitution method).